src/module.cpp

```diff
@@ -2417,7 +2417,7 @@ int Module::execPreprocessor(const char *infilename, llvm::raw_string_ostream *o
     else
         opts.addMacroDef("ISPC_POINTER_SIZE=64");
 
-    if (g->target->hasHalf())
+    if (g->target->hasHalfConverts())
         opts.addMacroDef("ISPC_TARGET_HAS_HALF");
```

Comment:
I would rename the macro `ISPC_TARGET_HAS_HALF` to `ISPC_TARGET_HAS_HALF_CONVERSION` to avoid possible confusions with `ISPC_FP16_SUPPORTED`

Reply:
`ISPC_TARGET_HAS_HALF ` is not mentioned in our docs, so it's probably ok to remove it. But i don't want to change it for compatibility reasons.